### PR TITLE
Added week9 solutions

### PR DESCRIPTION
**Pull request by mdreaj38**

### CODE REVIEW

File: week09/NumberOfIsland.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// TC: O(MN)
+// MC: O(MN)
+class Solution {
+public:
+	static const char DONE = '#';
+	static constexpr int dRow[4] = {1, -1, 0, 0};
+	static constexpr int dCol[4] = {0, 0, 1, -1};
```
Comment (Sadman007) [inline week09/NumberOfIsland.cpp:7]: Nice

File: week09/time-needed-to-inform-all-employees.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		for(int i : adjList[root]){
+			maxTime = max(maxTime, findMinTime(i, informTime));
+		}
+
+		return maxTime + informTime[root];
+	}
+
+	int numOfMinutes(int n, int headID, vector<int>& manager, vector<int>& informTime) {
+        adjList.clear();
+		adjList.resize(n);
```
Comment (Sadman007) [inline week09/time-needed-to-inform-all-employees.cpp:19]: This is actually very risky! Resize does not CLEAR the old data. So if you do the following operations in order, you will get error:

```c++
Solution sol = new Solution();
cout << sol.numOfMinutes(n1, head1, manager1, informTime1) << "\n";
cout << sol.numOfMinutes(n2, head2, manager2, informTime2) << "\n"; // sol will contain the older data from the first operation
```

So, clear the adjList always.

**Review by Sadman007 — APPROVED** [review]
Nice work!

File: week09/time-needed-to-inform-all-employees.cpp (hunk 2 above)
Comment (mdreaj38) [reply week09/time-needed-to-inform-all-employees.cpp:19]: vaia its already cleared at 18th line

**Pull request merged**